code/modules/aquarium/fish.dm

```diff
@@ -0,0 +1,139 @@
+	return pickweight(probability_table)
+
+// Actual fish definitions below - there's no specific paths, they are autogenerated from behaviours
+
+// Freshwater fish
+
+/datum/aquarium_behaviour/fish/goldfish
+	name = "goldfish"
+	desc = "not actually made of gold"
+	icon_state = "goldfish"
+	color = "#f2da29"
+	aquarium_vc_color = "#f2da29"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+
+/datum/aquarium_behaviour/fish/angelfish
+	name = "angelfish"
+	desc = "Omnivore fish originating from Amazon River."
+	icon_state = "angelfish"
+	dedicated_in_aquarium_icon_state = "bigfish"
+	sprite_height = 7
+	source_height = 7
+
+/datum/aquarium_behaviour/fish/guppy
+	name = "guppy"
+	desc = "One of the most popular aquarium fish."
+	icon_state = "guppy"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+	aquarium_vc_color = "#91AE64"
+	sprite_width = 8
+	sprite_height = 5
+
+/datum/aquarium_behaviour/fish/plasmatetra
+	name = "plasma tetra"
+	desc = "Tetra fish with coloring similar to plasma."
+	icon_state = "plastetra"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+	aquarium_vc_color = "#D30EB0"
+
+/datum/aquarium_behaviour/fish/catfish
+	name = "cory catfish"
+	desc = "Unrelated to actual cats."
+	icon_state = "catfish"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+	aquarium_vc_color = "#907420"
+
+/datum/aquarium_behaviour/fish/spacecarp
+	name = "space carp"
+	desc = "This space predator fish is known to cause yearly damage to space stations during their migrations."
+	icon_state = "carp"
+	sprite_width = 8
+	sprite_height = 8
+	availible_in_random_cases = FALSE
+
+// Saltwater fish below
+
+/datum/aquarium_behaviour/fish/clownfish
+	name = "clownfish"
+	desc = "Honk!"
+	icon_state = "clownfish"
+	dedicated_in_aquarium_icon_state = "clownfish_small"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+	sprite_width = 8
+	sprite_height = 5
+
+/datum/aquarium_behaviour/fish/cardinal
+	name = "cardinalfish"
+	desc = "Small saltwater fish active at night."
+	icon_state = "cardinalfish"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+
+/datum/aquarium_behaviour/fish/greenchromis
+	name = "green chromis"
+	desc = "Small fish with distinctive green sheen."
+	icon_state = "greenchromis"
+	dedicated_in_aquarium_icon_state = "fish_greyscale"
+	aquarium_vc_color = "#00ff00"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+
+/datum/aquarium_behaviour/fish/firefish
+	name = "firefish goby"
+	desc = "Long thing two-colored fish."
+	icon_state = "firefish"
+	sprite_width = 6
+	sprite_height = 5
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+
+/datum/aquarium_behaviour/fish/pufferfish
+	name = "pufferfish"
+	desc = "I'll let reviewers write these dang descriptions."
+	icon_state = "pufferfish"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+	sprite_width = 8
+	sprite_height = 8
+
+/datum/aquarium_behaviour/fish/pufferfish
+	name = "lanternfish"
+	desc = "Terror of the deep."
```

Comment:
"We lose approximately 30 moth crewmembers to these a shift at our underwater stations."